### PR TITLE
Removes and suppresses compiler warnings in static analyzer tests

### PR DESCRIPTION
**Pull request by t-rasmud**

### CODE REVIEW

**Comment by t-rasmud** [comment]
@swift-ci please test

**Comment by jkorous-apple** [comment]
@swift-ci test

**Review by jkorous-apple — APPROVED** [review]

File: clang/test/Analysis/region-store.c [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // RUN: %clang_analyze_cc1 -analyzer-checker=core,unix,debug.ExprInspection \
 // RUN:    -verify -analyzer-config eagerly-assume=false -std=c99 %s \
-// RUN:    -Wno-implicit-function-declaration
+// RUN:    -Wno-implicit-function-declaration -w
```
Comment (haoNoQ) [inline clang/test/Analysis/region-store.c]: `-Wno-implicit-function-declaration` can now be dropped, it's implied by `-w` ^^

**Review by haoNoQ — APPROVED** [review]

**Comment by t-rasmud** [comment]
@swift-ci test macOS

**Comment by t-rasmud** [comment]
@swift-ci test

**Review by haoNoQ — APPROVED** [review]
LGTM thanks!!

**Comment by t-rasmud** [comment]
@swift-ci test macOS

**Pull request merged**